serve_test.go

```diff
@@ -0,0 +1,58 @@
+package goss
+
+import (
+	"bytes"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"testing"
+
+	"github.com/aelsabbahy/goss/util"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestServe(t *testing.T) {
+	t.Parallel()
+	tests := map[string]struct {
+		specFile           string
+		expectedHTTPStatus int
+	}{
+		"passing": {
+			specFile:           filepath.Join("testdata", "passing.goss.yaml"),
+			expectedHTTPStatus: http.StatusOK,
+		},
+		"failing": {
+			specFile:           filepath.Join("testdata", "failing.goss.yaml"),
+			expectedHTTPStatus: http.StatusServiceUnavailable,
+		},
+	}
+	for testName := range tests {
+		tc := tests[testName]
+		t.Run(testName, func(t *testing.T) {
+			var logOutput bytes.Buffer
+			log.SetOutput(&logOutput)
+
+			config, err := util.NewConfig(util.WithSpecFile(tc.specFile))
+			require.NoError(t, err)
+			t.Logf("Config: %v", config)
```

Comment:
It's handy to see (on test-failure only) the process-level goss configuration, even if it takes a little work to eyeball-parse.